core/src/toga/widgets/mapview.py

```diff
@@ -185,29 +186,34 @@ def location(self, coordinates: toga.LatLng | tuple[float, float]):
     def zoom(self) -> int:
         """Set the zoom level for the map.
 
-        The zoom level is an integer in the range 0-5 (inclusive). The level of detail
-        visible at each zoom level is:
+        The zoom level is an integer in the range 0-18 (inclusive). It can be used to
+        set the number of degrees of longitude that will span a 256 CSS pixel region the
+        vertical axis of the map, following the relationship::
+
+            visible_longitude = 180 / (2**(zoom + 8))
```

Comment:
This was some implementation drift on my part. My first pass implementation on macOS was using the size of the viewable window, not a 256px tile (and even then, my math was out, because I was using longitude, not latitude at the equator).

Reply:
I'm glad you poked at this, because the math was *way* off on macOS/iOS. I was using longitude span, not latitude span, and there needs to be an additional factor accounting for the display scale factor to make the zoom factors line up with OSM. The math should now be correct; I've also compared the maps at the same zoom level and window size between macOS, iOS and Android, and they produce approximately the same viewable area for the same zoom level.

As part of that - I'm pretty sure it's `latitude_per_256 = 180/2**zoom`. The `2**zoom` factor is the divisor for the entire possible area, which is 180 degrees on the *vertical* calculation. The math on the OSM reference page is computing the distance in the *horizontal* direction, which is twice as wide (because there's 360 degrees in the equator).

Reply:
Strike that - it *is* 360. I was being thrown off by the fact that the scale factor on retina displays is 2. CI uses a non-retina display (so - scale factor 1), which revealed the problem. Some iOS devices (more recent ones) have scale factors of 3, which doubly reinforced the point.

Reply:
I think a link to the OpenStreetMap page would be useful in case the reader wants more detail.

Also, if you look at the map on that page, you can see why they chose to define it in terms of longitude, because in the Mercator projection, which macOS also uses, the longitude has a constant degrees per pixel all over the world, but the latitude doesn't.

Reply:
I think a link to the OpenStreetMap page would be useful in case the reader wants more detail.

Also, if you look at the map on that page, you can see why they chose to define it in terms of longitude, because in the Mercator projection, which macOS also uses, the longitude has a constant degrees per pixel all over the world, but the latitude doesn't.